### PR TITLE
Fixed visible line ripple when outlined and outerStyle in case of no helper text is shown

### PR DESCRIPTION
**Pull request by Cromefire**
fixed that the line below the text field is shown when outlined is activated
screenshot from 2018-07-07 12-37-58

also outerStyle right now just works with helper text visible

### CODE REVIEW

**Comment by artipost** [comment]
Failed pictures (1): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/6dd2117ce3f6507a872d71d8e7229ffd/raw/2c7b8ec70b53fa634f5c6ed1e57dcc56ee1266a2/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [401187426](https://travis-ci.org/prateekbh/preact-material-components/builds/401187426), for commit [bfaad93](https://github.com/prateekbh/preact-material-components/pull/856/commits/bfaad93dc9b31c0e6995a6eb444943f6bd0341bc).

**Comment by Cromefire** [comment]
Just confirmed that it is working correctly with this pr

**Comment by Cromefire** [comment]
Did not see this:
```
return showDiv ? (
       ^^^^^^^
```

**Comment by Cromefire** [comment]
Now fixed

**Comment by artipost** [comment]
Failed pictures (1): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/98c7bf0dc51ba8f6e494e6c843596490/raw/11cc04cb6e72768d34ab6baa7785224cd0c9a215/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [401197756](https://travis-ci.org/prateekbh/preact-material-components/builds/401197756), for commit [0897477](https://github.com/prateekbh/preact-material-components/pull/856/commits/0897477bcbed2f582973d3987d6254dc8a65f364).

**Comment by artipost** [comment]
Failed pictures (1): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/9b128ecc77cf5650a1ce3012e30fbee1/raw/31749641f47b2f63193f529a6e726336ef98158d/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [401200051](https://travis-ci.org/prateekbh/preact-material-components/builds/401200051), for commit [d4df18f](https://github.com/prateekbh/preact-material-components/pull/856/commits/d4df18fc39ed210238085918c4b3fa5311eefd3f).

**Comment by Cromefire** [comment]
The rotation doesn't get applied in the docs, but for me it is working I'll investigate it, but generally it should work fine now

**Comment by artipost** [comment]
Failed pictures (1): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/581378aab53e6aba9756aa8db43e1f64/raw/318c046d731fd426b3a694731087d34c979302c2/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [401203654](https://travis-ci.org/prateekbh/preact-material-components/builds/401203654), for commit [5f065ac](https://github.com/prateekbh/preact-material-components/pull/856/commits/5f065ac953cf3020de2f24227f0405da9ceaed60).

**Comment by artipost** [comment]
Failed pictures (1): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/80d80bca907bb3f4d9b67bbbf921d713/raw/6a70ddb908848d9504e7d9aa063f508493a33e02/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [401204898](https://travis-ci.org/prateekbh/preact-material-components/builds/401204898), for commit [5ca412c](https://github.com/prateekbh/preact-material-components/pull/856/commits/5ca412c98697e3701eb3987a1f86f76cd47edbea).

**Comment by Cromefire** [comment]
Ready for merging

**Comment by artipost** [comment]
Failed pictures (1): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/55ecd6a78829d520f3c6150d657ac9a6/raw/1bf25c764a15ed04b91173229ece5617c18b39db/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [401467552](https://travis-ci.org/prateekbh/preact-material-components/builds/401467552), for commit [a535142](https://github.com/prateekbh/preact-material-components/pull/856/commits/a53514218db8d55e3690b34de46c956e34c22789).

**Comment by artipost** [comment]
Failed pictures (1): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/ed1dd7f1095b65bcea5c96de3c991c9f/raw/ec9c5eee52ba540b28892168bec2e0fe38f8ab75/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [401490262](https://travis-ci.org/prateekbh/preact-material-components/builds/401490262), for commit [18ee25b](https://github.com/prateekbh/preact-material-components/pull/856/commits/18ee25b415b0a2277aa114cda7490da57c08d565).

**Comment by artipost** [comment]
Failed pictures (1): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/533c9231e8a448d0d7cc4e351b034e96/raw/61ddfd216c5a7137dbfde75a4024af1279587a96/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [401490594](https://travis-ci.org/prateekbh/preact-material-components/builds/401490594), for commit [8701e43](https://github.com/prateekbh/preact-material-components/pull/856/commits/8701e43ddd228c4868b5aa8708a28b73b00c7653).

**Comment by artipost** [comment]
Failed pictures (1): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/2b2466ac2d8f8dc0a4232d0969a9d9ec/raw/ec79180fc87a444759d49306c6fc1597f652e1ec/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [401492079](https://travis-ci.org/prateekbh/preact-material-components/builds/401492079), for commit [7eab1ce](https://github.com/prateekbh/preact-material-components/pull/856/commits/7eab1ce33493a204d1981aa5d9fb1a36cdbc5ccb).

**Comment by Cromefire** [comment]
Ready for merging it's just select test failing, should I fix the test?

**Comment by Cromefire** [comment]
Select is somehow rendered 2x under each other

**Comment by prateekbh** [comment]
@Cromefire yeah thats an infra bug, dont worry about it i'll merge once i review

File: TextField/TextField.jsx [hunk 1]
```diff
@@ -199,6 +205,13 @@ class TextField extends Component {
       props.id = 'tf-' + this.id;
     }
 
+    const intermediateStyle = Object.assign(
```
Comment (prateekbh) [inline TextField/TextField.jsx]: why this?

**Comment by Cromefire** [comment]
Okay I will remove that, then people will have to suggest new defaults for the intermediate container (if you apply width and there isn't that default 100% it won't sizeup the actual input) or use classes.

File: TextField/TextField.jsx (hunk 1 above)
Comment (Cromefire) [reply TextField/TextField.jsx]: Else when scaling up the intermediate layer (the actual input, when using helper text) won't scale up, when scaling up the outer `<div>`.
```
__________________________________
|  |-----------|                 |
|  | input     |  outer div      |
|  |-----------|                 |
|  <helpertext>                  |
__________________________________
```

Comment (Cromefire) [reply TextField/TextField.jsx]: That my personal use case, why I added the outerStyle

**Comment by artipost** [comment]
Failed pictures (1): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/ba2a46b2f8d90a15cef4769287f716fd/raw/5d183cb8dccfbbd1160d57e9f6d23a8ff6d7910a/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [401733531](https://travis-ci.org/prateekbh/preact-material-components/builds/401733531), for commit [cc3ecb7](https://github.com/prateekbh/preact-material-components/pull/856/commits/cc3ecb7e98d91bb1fe14d4281cb0bc05fd22c84f).

File: TextField/TextField.jsx [hunk 2]
```diff
@@ -134,19 +135,22 @@ class TextFieldInput extends MaterialComponent {
...
             </svg>
           </div>
         ) : null}
-        {props.outlined ? <div className="mdc-notched-outline__idle" /> : null}
+        {outlined ? <div className="mdc-notched-outline__idle" /> : null}
       </div>
     );
   }
 }
+TextFieldInput.defaultProps = {
```
Comment (prateekbh) [inline TextField/TextField.jsx]: is this needed anymore?

**Comment by Cromefire** [comment]
To mitigate this:
```
__________________________________
|  |-----------|                 |
|  | input     |  outer div      |
|  |-----------|                 |
|  <helpertext>                  |
__________________________________
```
I have defined a default for width (100%), but I can't define defaults for any CSS property.

But you are right that should be through CSS classes

**Comment by prateekbh** [comment]
I can see why fixing visible line ripple is needed but not for intermediateStyle..
can you break the PR into two, if you still have a solid use case?

**Comment by Cromefire** [comment]
intermediateStyle is removed

File: TextField/TextField.jsx (hunk 2 above)
Comment (Cromefire) [reply TextField/TextField.jsx]: Removed

**Comment by Cromefire** [comment]
Now the unneeded default props you mentioned are also removed

**Comment by Cromefire** [comment]
There is just the default intermediateStyle left, which is for convenience so not everyone has to define things like `width: 100%` on it, can be expanded for other CSS props.

**Comment by Cromefire** [comment]
So I changed what you requested, please review again

**Comment by Cromefire** [comment]
I also verified that everything is working (using the local documentation server) this time.

**Comment by Cromefire** [comment]
screenshot_2018-07-09 preact-material-components-site

**Comment by prateekbh** [comment]
Thanks, this release might be a lil delayed, may be EOW. as there is also a new mdc-web release

File: TextField/TextField.jsx [hunk 3]
```diff
@@ -199,6 +200,10 @@ class TextField extends Component {
       props.id = 'tf-' + this.id;
     }
 
+    const intermediateStyle = {
```
Comment (prateekbh) [inline TextField/TextField.jsx]: delete this

File: TextField/TextField.jsx [hunk 4]
```diff
@@ -212,6 +217,7 @@ class TextField extends Component {
             <label for={props.id}>{props.cssLabel || `${props.label}: `}</label>
           )}
         <TextFieldInput
+          outerStyle={intermediateStyle}
```
Comment (prateekbh) [inline TextField/TextField.jsx]: delete this

File: TextField/TextField.jsx [hunk 5]
```diff
@@ -228,6 +234,7 @@ class TextField extends Component {
       <TextFieldInput
         {...props}
         className={className}
+        outerStyle={outerStyle}
```
Comment (prateekbh) [inline TextField/TextField.jsx]: delete this, outer style should only apply to outermost container

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "preact-material-components",
-  "version": "1.4.7",
+  "version": "1.4.8",
```
Comment (prateekbh) [inline package.json]: revert this, next mdc-web release will bump this to 1.5.0

File: TextField/index.d.ts [hunk 7]
```diff
@@ -15,6 +15,7 @@ declare interface ITextFieldProps extends JSX.HTMLAttributes {
   box?: boolean;
   outlined?: boolean;
   helperText?: string;
+  intermediateStyle?: {[key: string]: string};
```
Comment (Cromefire) [inline TextField/index.d.ts]: delete this

File: TextField/TextField.jsx (hunk 5 above)
Comment (Cromefire) [reply TextField/TextField.jsx]: In that case this is the outermost container

File: TextField/TextField.jsx (hunk 3 above)
Comment (Cromefire) [reply TextField/TextField.jsx]: That are the defaults, I would like to keep them. You can't see them from the outside, but if you are really sure, I can remove them

File: TextField/TextField.jsx [hunk 8]
```diff
@@ -47,6 +47,7 @@ const defaultProps = {
  * @prop outlined = false
  * @prop box = false
  * @prop type = 'text'
+ * @prop outerStyle = {[key: string]: string}
```
Comment (Cromefire) [inline TextField/TextField.jsx]: change to "= {}"

**Comment by Cromefire** [comment]
Okay I can understand that

**Comment by artipost** [comment]
Failed pictures (1): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/561fe37e7901bd4e34a8cd56e94c7234/raw/95b9e3b909379e74b794c41ff9ceaeebe2a38e9d/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [402378461](https://travis-ci.org/prateekbh/preact-material-components/builds/402378461), for commit [11124c7](https://github.com/prateekbh/preact-material-components/pull/856/commits/11124c78bdad4f87d042ab858841521e1afa38ac).

**Comment by Cromefire** [comment]
I will investigate the wired failure of the documentation check, I can't see any differences

**Comment by Cromefire** [comment]
I found that the text field now also has the this dublication bug, I think that's up to you.

**Comment by prateekbh** [comment]
@Cromefire the duplicate screen bug is everywhere, just download the new pic and put it in the golden tests folder.
I'll release today

**Comment by Cromefire** [comment]
Okay I'll do that

**Comment by Cromefire** [comment]
Should be it

**Comment by prateekbh** [comment]
Thanks for all the contribution, its highly appreciated.
I'll release it today.

**Comment by Cromefire** [comment]
May I ask when 1.5.0 is released?

**Comment by prateekbh** [comment]
done

**Comment by cromefire** [comment]
Thanks

**Pull request merged**